controllers/apicast_controller.go

```diff
@@ -233,7 +234,7 @@ func (r *ApicastReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 // SetupWithManager sets up the controller with the Manager.
 func (r *ApicastReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&saasv1alpha1.Apicast{}).
+		For(&saasv1alpha1.Apicast{}, builder.WithPredicates(util.ResourceGenerationOrFinalizerChangedPredicate{})).
```

Comment:
I found out that the operator-utils lib we are using has this small helper to filter out events when status or metadata is changed in custom resources, which avoids unnecessary reconcile loops. I added it to all controllers.